src/main/java/ui/GUIController.java

```diff
@@ -25,8 +26,21 @@
     private final PanelControl panelControl;
     private final UI ui;
     private final Label apiBox;
+
+    /**
+     * To store the previous amount of the available remaining API request.
+     */
+    private int previousRemainingApiRequests = 0;
```

Comment:
yea. If i remove previous. This statement will look weird.
`int difference = previousRemainingApiRequests - remainingRequests;`